### ISSUE TITLE
Editior Base Font shown as "(null) -0.0"

### ISSUE THREAD

**Issue by gongjue** [MacDownApp/macdown#670]
The font of the editor cannot be changed, and in the preference panel the Base Font is shown as "(null) - 0.0". If I click "change" button and select one font, nothing happens. Currently the font of the editor is Arial, the default font. Does anyone know why this happens, or which file controls this setting?

<img width="547" alt="screen shot 2016-10-05 at 11 55 34" src="https://cloud.githubusercontent.com/assets/6719802/19127154/af14c872-8af2-11e6-8095-e860453175cf.png">

**Comment by franklinyu** [comment]
The preference of MacDown is saved in these two files:

```shell
$HOME/Library/Preferences/com.uranusjr.macdown.plist
$HOME/Library/Preferences/com.uranusjr.macdown.LSSharedFileList.plist
```

Try moving these preferences to somewhere else (like `/tmp`), then restart MacDown. It would be appreciated if you would provide the corrupted preference file, if the issue disappear after resetting the preference, so that we can debug on that to prevent it from happening again.

**Comment by gongjue** [comment]
Based on the information provided by @franklinyu , I finally figured out the problem. I deleted the font set as the base font of MacDown base font, than the app cannot find the font and indicates it as 'null'. The solution is edit the attribute `editorBaseFontInfo` in `$HOME/Library/Preferences/com.uranusjr.macdown.plist` to an existing font.

**Comment by franklinyu** [comment]
Thank you for your feedback. Did you tweak the plist yourself (with plist editors)? If the missing `editorBaseFontInfo` is caused by actions in preference panel only, then this should be fixed. A normal user should not be able to corrupt the preference file.

If you did tweak the plist yourself, I'll close this issue since it is solved.

**Comment by gongjue** [comment]
Yes, I edit the plist by Xcode. The issue happened because I set the Base Font and later deleted that specific font from the system; the attribute of `editorBaseFontInfo` in the plist is the font I set before but cannot be traced in the system.

**Comment by franklinyu** [comment]
So you mean that the font is set to "FooBarFont" before you uninstalled "FooBarFont" manually? I think it is possible for a non-technician to encounter this situation, so I guess MacDown should have provided more information (such as "Oh, I can't find the font you specified") to the end-user, rather than simply displaying "null - 0.0" which does not provide any useful information to a non-tech. I will mark this issue as a feature request.

**Comment by gongjue** [comment]
Yes, exactly as what you described. If the app cannot find the font, a better way might be inform the user and provide a dialogue to select other fonts.

**Comment by uranusjr** [comment]
Maybe we can just fallback to the default font? That would be a lot simpler, and still makes sense (to me, at least).

**Comment by franklinyu** [comment]
@uranusjr So the default font cannot be manually deleted by user? Or you would like to keep a backup in the application?

**Comment by gongjue** [comment]
The font do fall back to a default one, in my case it goes to Arial. However in this situation the user can still click the "change" button on the right and select some font in the pop up dialogue window, but this has no effect on the font.

**Comment by uranusjr** [comment]
@franklinyu I think the default font can’t be deleted by the user unless you are determined to mess with the operating system really hard. A lot of things will be broken if you do.

@gongjue What I meant is that, MacDown should select a default font on its own when the user deletes the font it’s currently using. It should display it in the preferences, use it in the editor, and allow the user to later change it without errors.

**Comment by franklinyu** [comment]
@uranusjr Oh yes I got it. I just tried to delete a system font with [Font Book](https://support.apple.com/en-us/HT201749) and I found that the menu item "remove this font" is grayed out (I didn't expect this). So yes, I can confirm that system fonts cannot be deleted without hack. Therefore it is appropriate to set a system font as the default, exactly as you described.

**Comment by movingobjects** [comment]
I had this issue as well. I accidentally removed my Base Font, and then this font selection preference remained stuck on `(null) -0.0` even when trying to select a new one.

Editing the plist file worked to get me running again, but definitely would be great to have a fix that doesn't require editing that file to select a new font.
